cranelift/codegen/meta/src/shared/settings.rs

```diff
@@ -264,5 +264,13 @@ pub(crate) fn define() -> SettingGroup {
         true,
     );
 
+    // Spectre options.
+
+    settings.add_bool(
+        "enable_heap_access_spectre_mitigation",
+        "Enable Spectre mitigation on heap bounds checks.",
+        false,
```

Comment:
The risk of causing harm by disabling Spectre mitigations by default seems much worse than the performance hit that will be caused by its being set all the time, so I think we should enable it by default, explicit the performance overhead here, and say that while it's enabled by default, there are certain environments in which it's fine to disable it.

Reply:
See below: will need to be `false` by default unless we also implement for the legacy x86 backend (or until we delete it), which (unless I'm mistaken) seems to be missing some of the necessary bits for cmovs.

Reply:
Now on by default with latest version.